### PR TITLE
feature: modular top-bar items

### PR DESCRIPTION
**Pull request by inureyes**
This PR refactors top bar items in WebUI app shell to make app shell lighter / modular.

Also, it includes
 - Tab focus move between session launcher resource sliders.
 - Disable slider reset after changing cluster mode.

### CODE REVIEW

File: src/components/backend-ai-user-dropdown-menu.ts [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+    return html`
+      <div class="horizontal flex center layout">
+        <div class="vertical layout center" style="position:relative;right:50px;">
+          <mwc-menu id="dropdown-menu" class="user-menu">
+            ${this.domain !== 'default' && this.domain !== '' ? html`
+            <mwc-list-item class="horizontal layout start center" disabled style="border-bottom:1px solid #ccc;">
+                ${this.domain}
+            </mwc-list-item>
+            ` : html``}
+            <mwc-list-item class="horizontal layout start center">
```
Comment (agatha197) [inline src/components/backend-ai-user-dropdown-menu.ts]: This item only shows the user name. But this button is highlighted when hovering. So what do you think about adding `pointer-events:none` to reduce confusion??

File: src/components/backend-ai-project-switcher.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+ <backend-ai-project-switcher></backend-ai-project-switcher>
+
+ @group Backend.AI Web UI
+ @element backend-ai-project-switcher
+ */
+@customElement('backend-ai-project-switcher')
+export default class BackendAIProjectSwitcher extends LitElement {
+  @property({type: Array}) groups = [];
+  @property({type: String}) current_group = '';
```
Comment (agatha197) [inline src/components/backend-ai-project-switcher.ts]: How about use camelCase?

File: src/components/backend-ai-user-dropdown-menu.ts [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+ Example:
+
+ <backend-ai-user-dropdown-menu></backend-ai-user-dropdown-menu>
+
+ @group Backend.AI Web UI
+ @element backend-ai-user-dropdown-menu
+ */
+@customElement('backend-ai-user-dropdown-menu')
+export default class BackendAiUserDropdownMenu extends LitElement {
+  @property({type: String}) user_id = 'DISCONNECTED';
```
Comment (agatha197) [inline src/components/backend-ai-user-dropdown-menu.ts]: Same as above.

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]
Everything works well. But I suggest some minor changes such as unify naming convention (camelCase), and add tab focus move to cluster slider too.

when moving to **resource slider** using the tab.
image

cluster
image

File: src/components/backend-ai-user-dropdown-menu.ts (hunk 1 above)
Comment (inureyes) [reply src/components/backend-ai-user-dropdown-menu.ts]: @Sujin-Kim1 reflected.

File: src/components/backend-ai-project-switcher.ts (hunk 2 above)
Comment (inureyes) [reply src/components/backend-ai-project-switcher.ts]: @Sujin-Kim1 it has a history. In Polymer, attributes with `-` will convert to automatic camelCase property. Now we use Lit, therefore it is ok to change them to camelCase. I'll reflect them.

Comment (inureyes) [reply src/components/backend-ai-project-switcher.ts]: https://github.com/lit/lit-element/issues/29

File: src/components/backend-ai-user-dropdown-menu.ts (hunk 3 above)
Comment (inureyes) [reply src/components/backend-ai-user-dropdown-menu.ts]: Resolved.

File: src/components/backend-ai-project-switcher.ts (hunk 2 above)
Comment (inureyes) [reply src/components/backend-ai-project-switcher.ts]: Resolved.

Comment (agatha197) [inline src/components/backend-ai-project-switcher.ts]: Oops, I found another one that doesn't fit the naming convention.

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]

File: src/components/backend-ai-project-switcher.ts (hunk 2 above)
Comment (inureyes) [reply src/components/backend-ai-project-switcher.ts]: Updated.

**Review by Sujin-Kim1 — APPROVED** [review]
LGTM!

**Pull request merged**